provenance/provenance.sol

```diff
@@ -0,0 +1,55 @@
+// provenance concept contract created by M.ElBoudi, A.Blackwell, M.Terzi et al during Ethereum development workshop in London
+
+contract Certifiers {
+        mapping (address => bool) certifier;
+        address provenance;
+        
+        //initialiser is run the first time the contract is uploaded to the network. Launched by Provenance and reads provenance "public key"
+        function Certifiers () {
+            provenance = msg.sender;    
+        }
+        
+        function createNewCertifier() returns (bool completed) {
+                certifier[msg.sender] = false;
```

Comment:
I'm not sure if I understand this correctly. Every key exists in a mapping from the beginning and is initialised to "zero", e.g. `false` for a `bool`. So this is more like `resetCertify`, i.e. it only reverses the effect of `certify`.